src/util/parse_lykeion.rs

```diff
@@ -1,36 +1,33 @@
 use chrono::NaiveDate;
 use regex::Regex;
 use rss::Channel;
-
 lazy_static::lazy_static! {
     static ref FOOD_REGEX: Regex = Regex::new(r"Lounas\s:\s([\W\w]+)?<br>").unwrap();
     static ref ALLERGEENI_REGEX: Regex = Regex::new(r"\s\(.*?\)").unwrap();
 }
-
 pub async fn parse_lykeion() -> Result<Vec<(NaiveDate, String)>, anyhow::Error> {
     let r_client = reqwest::Client::builder().build().unwrap();
     let mut retvec = Vec::new();
     for i in 1..=7 {
         let resp = r_client
             .get(format!("https://aromimenu.cgisaas.fi/VaasaAromieMenus/fi-FI/Default/_/CampusLykeion/Rss.aspx?Id=0c9160c7-bedb-4b60-9ee1-188bf43a02b3&DateMode={}", i))
-            .send()
-            .await?
-            .bytes()
-            .await?;
+            .send().await?
+            .bytes().await?;
         let channel = Channel::read_from(&resp[..])?;
         for item in channel.items {
-            let date_str = item.title.unwrap();
-            let date_str = &date_str[date_str.find(' ').unwrap()+1..];
+            let date_str = &item.title.unwrap()[date_str.find(' ').unwrap()+1..];
```

Comment:
I don't think this is how it works...

Reply:
yesyes that was actually only one I was unsure about

Reply:
Why tf won't it work though

Reply:
you are referencing date_str before it is assigned (as `date_str.find(...`)

Reply:
yes I figured it out

Reply:
I didn't see that other `date_str` there